### PR TITLE
[master] Add syncing of custom ssh wrappers

### LINKED ISSUE #45450: Allow custom modules which run on master
### Setup

I am using `salt-ssh` to deploy configuration to servers. On my computer (SSH master) I have in `states` directory also `_modules` with some simple Python functions I want to run to process values in Jinja templates. I do something like:

```
{% for user in pillar.users %}
  {% set user_groups = salt['servers.groups'](user.get('servers', {}), grains['id']) %}
  ...
```

`servers.groups` is a Python function which returns values from user's pillars based on target ID.

The issue is that this simple pure function runs over SSH on the minion. This is problematic because it really slows down things. Is there a way to configure that the module should be only used during rendering of the state and that it can be run locally on master?

### Versions Report

```
Salt Version:
           Salt: 2016.11.1
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.14 (default, Sep 23 2017, 22:06:14)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.3
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.2
            ZMQ: 4.1.6
 
System Versions:
           dist: Ubuntu 17.10 artful
        machine: x86_64
        release: 4.13.0-21-generic
         system: Linux
        version: Ubuntu 17.10 artful
```

### PR DESCRIPTION
**Pull request by lkubb**
### What does this PR do?
Adds `sync_wrapper` to the `saltutil` runner/execution/state modules.
From what I can tell, the loader is already configured to load them.

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/45450
Reference: https://github.com/saltstack/salt/issues/61637

### Previous Behavior
`<cache_dir>/extmods/wrapper` is respected by the loader, but cannot be synced from the fileserver

### New Behavior
`<cache_dir>/extmods/wrapper` can be synced

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

File: salt/runners/saltutil.py [hunk 1]
```diff
@@ -835,3 +840,34 @@ def sync_executors(saltenv="base", extmod_whitelist=None, extmod_blacklist=None)
         extmod_whitelist=extmod_whitelist,
         extmod_blacklist=extmod_blacklist,
     )[0]
+
+
+def sync_wrapper(saltenv="base", extmod_whitelist=None, extmod_blacklist=None):
+    """
+    .. versionadded:: 3000
```
Comment (Ch3LL) [inline salt/runners/saltutil.py:847]:
```suggestion
    .. versionadded:: 3007.0
```

File: salt/states/saltutil.py [hunk 2]
```diff
@@ -349,3 +349,17 @@ def sync_serializers(name, **kwargs):
             - refresh: True
     """
     return _sync_single(name, "serializers", **kwargs)
+
+
+def sync_wrapper(name, **kwargs):
+    """
+    Performs the same task as saltutil.sync_wrapper module
```
Comment (Ch3LL) [inline salt/states/saltutil.py:356]:
```suggestion
    .. versionadded:: 3007.0
    
    Performs the same task as saltutil.sync_wrapper module
```

**Review by Ch3LL — CHANGES_REQUESTED** [review]
Could you add a test that syncs a custom salt-ssh wraper and then salt-ssh actually uses that wrapper?

Also, this is great, thanks once again.

**Comment by lkubb** [comment]
Whoops, I was a bit too hasty in creating this PR, sorry. It was probably the surprise to find that 95% of the functionality was there already...

`versionadded` is now corrected and I added a test for syncing + usage. It works as expected on my machine, yay.

**Review by Ch3LL — APPROVED** [review]

**Review by whytewolf — APPROVED** [review]

**Pull request merged**